pkg/registry/generic/etcd/etcd.go

```diff
@@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 		if newVersion != version {
 			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the resource was updated to %d", version))
+			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("another client made an update to this object in resource version %d", version))
```

Comment:
Not sure there is much value in the older value - either you set it and don't need it, or you didn't set it and so it means nothing to you.  The new value at least tells you what is on the server which doesn't require you potentially dropping context to run kubectl get foo bar.

> On Jun 10, 2015, at 4:58 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/registry/generic/etcd/etcd.go:
> 
> > @@ -301,7 +301,7 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
> >  		}
> >  		if newVersion != version {
> >  			// TODO: return the most recent version to a client?
> > -			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the resource was updated to %d", version))
> > +			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("another client made an update to this object in resource version %d", version))
> Not sure if I agree with giving the new resource version, because someone copy and pasting it is probably blowing away a change that they don't intend to.
> 
> —
> Reply to this email directly or view it on GitHub.
>